File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        int count=1,currNum = nums[0];
        
        for(int i=1;i<nums.size();i++){
            
            if(currNum != nums[i]){
                count--;
            }
            else
                count++;
            
            if(count == 0){
                count = 1;
                currNum = nums[i];
            }
        }
        
        return currNum;
    }
};

File: 2025-and-before/0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    
    let curr = nums[0];
    let count = 1;
    
    
    for(let i=1;i<nums.length;i++){
        
        if(curr == nums[i])
            count++;
        
        else count--;
        
        if(count == 0){
            count =1;
            curr =nums[i];
        }
    }
    
    return curr;
};